Mathlib/Data/List/Monad.lean

```diff
@@ -25,8 +25,8 @@ instance instMonad : Monad List.{u} where
 instance instLawfulMonad : LawfulMonad List.{u} := LawfulMonad.mk'
   (id_map := map_id)
   (pure_bind := fun _ _ => List.append_nil _)
-  (bind_assoc := List.flatMap_assoc)
-  (bind_pure_comp := fun _ _ => (map_eq_flatMap _ _).symm)
+  (bind_assoc := fun _ _ _ => List.flatMap_assoc)
```

Comment:
All those arguments should stay explicit in `flatMap_assoc` since it's a rewriting lemma